src/sage/rings/polynomial/multi_polynomial_ring_base.pyx

```diff
@@ -571,25 +571,22 @@ cdef class MPolynomialRing_base(CommutativeRing):
         else:
             return self._generic_coerce_map(self.base_ring())
 
-    cdef _coerce_c_impl(self, x):
+    cpdef _coerce_map_from_(self, other):
         """
-        Return the canonical coercion of x to this multivariate
-        polynomial ring, if one is defined, or raise a :exc:`TypeError`.
+        Return whether there is canonical coercion map
+        from the ring ``other`` to this multivariate polynomial ring.
 
         The rings that canonically coerce to this polynomial ring are:
 
         - this ring itself
-        - polynomial rings in the same variables over any base ring that
-          canonically coerces to the base ring of this ring
         - polynomial rings in a subset of the variables over any base ring that
           canonically coerces to the base ring of this ring
         - any ring that canonically coerces to the base ring of this polynomial
           ring.
 
         TESTS:
 
-        This fairly complicated code (from Michel Vandenbergh) ends up
-        implicitly calling ``_coerce_c_impl``::
+        Fairly complicated code (from Michel Vandenbergh)::
```

Comment:
Now that this is a `cpdef` method we can probably test it directly?

Reply:
I felt it's intended to be more of a "try to keep the test" than actually testing the method. But yes, `R.coerce_map_from(S)` should delegate to this function anyway.